### PR TITLE
Add instructions for building pyspark from source in readme

### PR DESCRIPTION
**Pull request by yangaws**
Commit 1: Add instructions for building pyspark from source in readme
Commit 2: Remove python 3.5 in tox to speed up testing
Commit 3: Remove unimplemented hyper-parameter in linear learner
Commit 4: Add instructions for s3 and s3a in README
Commit 5: Change jar path in index.rst

### CODE REVIEW

File: sagemaker-pyspark-sdk/src/sagemaker_pyspark/algorithms/LinearLearnerSageMakerEstimator.py [hunk 1]
```diff
@@ -516,13 +516,6 @@ class LinearLearnerBinaryClassifier(SageMakerEstimatorBase, LinearLearnerParams)
                              "recall_at_target_precision. Must be in range [0, 1]. ",
                              typeConverter=TypeConverters.toFloat)
 
-    positive_example_weight_mult = Param(Params._dummy(), "positive_example_weight_mult",
```
Comment (iquintero) [inline sagemaker-pyspark-sdk/src/sagemaker_pyspark/algorithms/LinearLearnerSageMakerEstimator.py]: this is an unrelated change to the readme. Not sure if it was intentional?

File: sagemaker-pyspark-sdk/tox.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = clean,flake8,docs,py{27,35,36},stats
+envlist = clean,flake8,docs,py{27,36},stats
```
Comment (iquintero) [inline sagemaker-pyspark-sdk/tox.ini]: you also probably didnt want to commit this.

File: sagemaker-pyspark-sdk/src/sagemaker_pyspark/algorithms/LinearLearnerSageMakerEstimator.py (hunk 1 above)
Comment (yangaws) [reply sagemaker-pyspark-sdk/src/sagemaker_pyspark/algorithms/LinearLearnerSageMakerEstimator.py]: Yep this is a typo we made before release. We removed setter and getter of this hyper-parameter but forgot to remove this declaration. This hyper-parameter is unimplemented by algorithm team. So setting it doesn't make any difference(but doesn't cause trouble either, just ignored by the image).

File: sagemaker-pyspark-sdk/tox.ini (hunk 2 above)
Comment (yangaws) [reply sagemaker-pyspark-sdk/tox.ini]: We need this to speed up integ tests. But I see your concern of 5 commits in PR.

Let me try rebasing to master. If not permitted, we can discuss on what to do next.
